static/js/components/CandidatesPreferencesForm.jsx

```diff
@@ -0,0 +1,637 @@
+    return formState.groupIDs.filter((value) => Boolean(value)).length >= 1;
+  };
+
+  const validateSorting = () => {
+    formState = getValues({ nest: true });
+    return (
+      // All left empty
+      (formState.sortingOrigin === "" &&
+        formState.sortingKey === "" &&
+        formState.sortingOrder === "") ||
+      // Or all filled out
+      (formState.sortingOrigin !== "" &&
+        formState.sortingKey !== "" &&
+        formState.sortingOrder !== "")
+    );
+  };
+
+  const onSubmit = async (formData) => {
+    const groupIDs = userAccessibleGroups.map((g) => g.id);
+    const selectedGroupIDs = groupIDs.filter(
+      (ID, idx) => formData.groupIDs[idx]
+    );
+    const data = {
+      id: Math.random().toString(36).substr(2, 5),
+      groupIDs: selectedGroupIDs,
+      name: formData.name,
+      savedStatus: formData.savedStatus,
+    };
+
+    if (addOrEdit === "Add") {
+      // Assign a random ID to the new profile
+      data.id = Math.random().toString(36).substr(2, 5);
+      data.default = true;
+    } else if (addOrEdit === "Edit") {
+      data.id = editingProfile?.id;
+      data.default = editingProfile?.default;
+    }
+
+    // decide if to show rejected candidates
+    if (formData.rejectedStatus) {
+      data.rejectedStatus = formData.rejectedStatus;
+    }
+    // Convert dates to ISO for parsing on back-end
+    if (formData.timeRange) {
+      data.timeRange = formData.timeRange;
+    }
+    if (formData.classifications.length > 0) {
+      data.classifications = formData.classifications;
+    }
+    if (formData.redshiftMinimum) {
+      data.redshiftMinimum = formData.redshiftMinimum;
+    }
+    if (formData.redshiftMaximum) {
+      data.redshiftMaximum = formData.redshiftMaximum;
+    }
+    if (formData.sortingOrigin) {
+      data.sortingOrigin = formData.sortingOrigin;
+      data.sortingKey = formData.sortingKey;
+      data.sortingOrder = formData.sortingOrder;
+    }
+
+    const currentProfiles = preferences.scanningProfiles || [];
+    if (addOrEdit === "Add") {
+      // Add new profile as the default in the preferences
+      currentProfiles.forEach((profile) => {
+        profile.default = false;
+      });
+      currentProfiles.push(data);
+    } else if (addOrEdit === "Edit") {
+      // Update profile
+      const profileIndex = currentProfiles.findIndex(
+        (profile) => profile.id === editingProfile?.id
+      );
+      if (profileIndex !== -1) {
+        currentProfiles[profileIndex] = data;
+      }
+    }
+
+    const prefs = {
+      scanningProfiles: currentProfiles,
+    };
+    dispatch(profileActions.updateUserPreferences(prefs));
+
+    if (addOrEdit === "Edit") {
+      // If we just edited the selected profile, let the
+      // parent component know we updated some fields
+      if (selectedScanningProfile?.id === data.id) {
+        setSelectedScanningProfile(data);
+      }
+      closeDialog();
+    } else if (addOrEdit === "Add") {
```

Comment:
@stefanv The randomly generated ids are used basically for internal identification of selected profiles, like here. They're mainly a holdover from the initial version of scanning profiles when I didn't ask the user for a name for the profile. I guess that could now all be replaced by checks against profile names if you'd prefer.

Reply:
If the profile name is required and is unique, that would make sense.  I'll leave this to your best design judgment.